packages/react-sdk/src/components/components-utils.ts

```diff
@@ -0,0 +1,86 @@
+import * as components from "./components";
+import { registeredComponents } from "./index";
+
+export type ComponentName = keyof typeof components;
+
+/**
+ * We need to define component names manually here, instead of using Object.keys(components)
+ * Otherwise every component would be included in the bundle, even if it is not used
+ */
+const componentNames = [
+  "Box",
+  "Body",
+  "TextBlock",
+  "Heading",
+  "Paragraph",
+  "Link",
+  "RichTextLink",
+  "Span",
+  "Bold",
+  "Italic",
+  "Superscript",
+  "Subscript",
+  "Button",
+  "Input",
+  "Form",
+  "Image",
+  "Blockquote",
+  "List",
+  "ListItem",
+] as const;
+
+type Equals<X, Y> = (<T>() => T extends X ? 1 : 2) extends <T>() => T extends Y
+  ? 1
+  : 2
+  ? true
+  : false;
+
+const typesEqual = <A, B>(a: Equals<A, B>) => undefined;
+
+/**
+ * Typecheck that
+ * - all component names are included in the componentNames array
+ * - there are no nonexistent component names in the componentNames array
+ **/
+typesEqual<ComponentName, typeof componentNames[number]>(true);
+
+export const getComponentNames = (): ComponentName[] => {
+  const uniqueNames = new Set([
+    ...componentNames,
+    ...Object.keys(registeredComponents || {}),
+  ]);
+
+  return [...uniqueNames.values()] as ComponentName[];
+};
+
+/**
+ * Now used only in builder app
+ * @todo Consider using the same approach in the builder app as in the published apps . A dynamic import is needed
+ */
+export const getComponent = (
+  name: string
+): undefined | typeof components[ComponentName] => {
+  return registeredComponents != null && name in registeredComponents
+    ? (registeredComponents[
+        name as ComponentName
+      ] as typeof components[ComponentName])
```

Comment:
😭 @Andarist any better ideas?

Reply:
The second cast is not needed as far as I can tell: [TS playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBKC2AHcBTMUByBDeKIwF4YBtAKBhgCIAhEAD0oBpyraATATyZYHoeqAKijpRqAGxDAA1twp8qACRRY2ASzABzWTHmUAClgBOWDccQALbboAy6mczn9KAJVXBzQkbbD3eTgMqIWGBWTrRibKFUAJJQWGJuUZT+AK6IKIYQwIaqiFBJqQBGWTl5SdQpUFDgSdFgiJVJAGIghvC18CYo5RLSAI4pIFDdDjpOttBJE1CxKO3MALowWPigkFCk8jAQWFCqEABmqngwhsps4GIcJADCIEio6Ni4jDAAdB93D2BomDgoxAWCwA3KRSFAOOkYM4UBp9sMzmwvsgfuh8EQDIY9vEADwAbz8MDwCXQAFo1DtCmIUKSfiJSSSUDAAAIQ9IlXJQUnE9RQHiFYKktl4FjEMD-GDqGDIx5-XALABcMDxAF9QSqAHyg0hraBwe4o37o5UsWh0JWq0bsDgWlWkO3gyFMmWouVMojCkAHfXfX7PPBilLwQoZBabfjCZBYmDC6UG2X+wgwKQoDhemNO9MIQ1o7WR1qwamwM5w6AZFBI+OuiBKmGlhEVl1GmAAHxgYBSYjESY7XbzdCjsF1sA0KCgTfQSYAFCxxbgldAcppSABKJUpMBsFBHH5sVsZ9JZqtG4gTt1LAgak0UM5QFKGMCnWHw8uV31omAAQiIve7ADI-3bCUpRLF9ETPCAWAoAB+GAp1AstwOPNEyAoNC0LnJkVjjd83SgtCFhXfClWzWUIDFCVsLPf0QXtYEgA)

You need to cast  `name` to `ComponentName` though as long as `name` is typed as `string` (and I assume that you want it to be typed as `string` because this is a dynamic piece)